automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -25,92 +31,112 @@ class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPA
+              120,
+              10 seconds,
+              s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish creating after 20 minutes"
+            )(implicitly, appInStateOrError(AppStatus.Running))
+            _ <- loggerIO.info(
+              s"AppCreationSpec: app ${googleProject.value}/${appName.value} monitor result: ${monitorCreateResult}"
             )
-            //IO(Sam.user.deleteResource("kubernetes-app", appName.value)(ronCreds.makeAuthToken()))
-          } else {
-            // Verify creating another app with the same disk doesn't error out
-            for {
-              _ <- loggerIO.info(
-                s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
+            _ = monitorCreateResult.status shouldBe AppStatus.Running
+
+            _ <- IO.sleep(1 minute)
+
+            // Delete the app
+            _ <- LeonardoApiClient.deleteApp(googleProject, appName, false)
+
+            // Verify getApp again
+            getAppResponse <- getApp
+            _ = getAppResponse.status should (be(AppStatus.Deleting) or be(AppStatus.Predeleting))
+
+            // Verify the app eventually becomes Deleted
+            // Don't fail the test if the deletion times out because the app pre-delete job can sporadically fail.
+            // See https://broadworkbench.atlassian.net/browse/IA-2471
+            listApps = LeonardoApiClient.listApps(googleProject, true)
+            implicit0(deletedDoneCheckable: DoneCheckable[List[ListAppResponse]]) = appDeleted(appName)
+            monitorDeleteResult <- streamFUntilDone(
+              listApps,
+              120,
+              10 seconds
+            ).compile.lastOrError
+            // TODO remove first case in below if statement when app deletion is reliable
+            _ <- if (!deletedDoneCheckable.isDone(monitorDeleteResult)) {
+              loggerIO.warn(
+                s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish deleting after 20 minutes. Result: $monitorDeleteResult"
               )
-              _ <- LeonardoApiClient.createAppWithWait(googleProject, restoreAppName, createAppRequest)(
-                client,
-                ronAuthorization,
-                loggerIO
-              )
-              _ <- LeonardoApiClient.deleteAppWithWait(googleProject, restoreAppName)
-            } yield ()
-          }
-        } yield ()
+              //IO(Sam.user.deleteResource("kubernetes-app", appName.value)(ronCreds.makeAuthToken()))
+            } else {
+              // Verify creating another app with the same disk doesn't error out
+              for {
+                _ <- loggerIO.info(
+                  s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
+                )
+                _ <- LeonardoApiClient.createAppWithWait(googleProject, restoreAppName, createAppRequest)(
```

Comment:
I'm having trouble understanding-- is the disk name to reuse passed in, or does the backend code automatically reuse a disk if one exists for that workspace/app combination?

Reply:
The name of the disk to be reused is passed in `createAppRequest` (the same request used to create the original instance of app). The new name of the app is passed through `restoreAppName`.